cfme/common/__init__.py

```diff
@@ -172,28 +173,33 @@ class TagPageView(BaseLoggedInPage):
     """Class represents common tag page in CFME UI"""
     title = Text('#explorer_title_text')
     table_title = Text('//div[@id="tab_div"]/h3')
-    tag_table = Table("//div[@id='assignments_div']//table")
-    tag_category = BootstrapSelect(id='tag_cat')
-    tag_name = BootstrapSelect(id='tag_add')
 
-    save_button = Button('Save')
-    cancel_button = Button('Cancel')
-    reset_button = Button('Reset')
+    @View.nested
+    class form(View):  # noqa
+        tags = Table("//div[@id='assignments_div']//table")
+        tag_category = BootstrapSelect(id='tag_cat')
+        tag_name = BootstrapSelect(id='tag_add')
+        entities = View.nested(BaseNonInteractiveEntitiesView)
+        save = Button('Save')
+        reset = Button('Reset')
+        cancel = Button('Cancel')
 
     @property
     def is_displayed(self):
         return (
             self.table_title.text == 'Tag Assignment' and
-            self.tag_table.is_displayed
+            self.form.tags.is_displayed
         )
 
 
 class WidgetasticTaggable(object):
     """
     This class can be inherited by any class that honors tagging.
-    Class should have 'EditTags' navigate step registered,
-    also 'Details' view should be present with tag attribute(widget with tag
-    information on details page)
+    Class should have following:
+        - 'Details' navigation
+        - 'Details' view should have entities.smart_management SummaryTable widget
+        - 'EditTags' navigation
+        - 'EditTags' view should have nested 'form' view with 'tags' table widget (TagPageView)
```

Comment:
@rlbabyuk thoughts on this wording, and the changes to TagPageView to match?

Reply:
As an option: 'TagPageView should be used in 'EditTags' navigation step.'

Reply:
yep, good point, I just put the class name in (), but it would be much more clear stated that way.